packages/client/src/objectSet/ObjectSetListenerWebsocket.ts

```diff
@@ -436,8 +457,42 @@ export class ObjectSetListenerWebsocket {
     const referenceUpdates = payload.updates.filter((update) =>
       update.type === "reference"
     );
+    const osdkObjectsWithReferenceUpdates = await Promise.all(
+      referenceUpdates.map(async (o) => {
+        const osdkObjectArray = await convertWireToOsdkObjects(
+          this.#client,
+          [{
+            __apiName: o.objectType,
+            __primaryKey: Object.values(o.primaryKey)[0],
```

Comment:
`o.primaryKey` is an object that contains a single mapping of the primary key property name to the property value.

Reply:
given we know what the primary key is from the metadata should we just do this directly instead of creating the temporary array to index into?

Reply:
sure, I can store the primary key name in the subscription name